runtimes/huggingface/tests/test_common.py

```diff
@@ -45,3 +47,48 @@ def test_load_pipeline(optimum_model: bool, expected):
     pipeline = load_pipeline_from_settings(hf_settings, model_settings)
 
     assert isinstance(pipeline.model, expected)
+
+
+@pytest.mark.parametrize(
+    "hf_pretrained_model, has_model_params, uri_value, expected",
+    [
+        (None, True, None, None),
+        (None, True, "", ""),
+        (None, True, "/some/folder/model-artefacts", "/some/folder/model-artefacts"),
+        (None, False, "N/A", None),
+        ("", True, None, None),
+        ("", True, "", ""),
+        ("", True, "/some/folder/model-artefacts", "/some/folder/model-artefacts"),
+        ("", False, "N/A", ""),
+        ("./some-pretrained-folder", "N/A", "N/A", "./some-pretrained-folder"),
```

Comment:
```suggestion
        ("./some-pretrained-folder", True, "N/A", "./some-pretrained-folder"),
        ("./some-pretrained-folder", False, "N/A", "./some-pretrained-folder"),
```
:wrench: The use of "N/A" for `has_model_params` breaks the type hint that this should be a Boolean if provided.

I'm also proposing two cases, so we can clearly see that the pre-trained URI takes precedence, even when the general URI is available.